### ISSUE TITLE
Transport protocols UDP/TLS/RTP/SAVPF and UDP/TLS/RTP/SAVP are not documented as valid

### ISSUE THREAD

**Issue by dilyanpalauzov** [sipwise/rtpengine#1595]
According to https://www.kamailio.org/docs/modules/devel/modules/rtpengine.html#rtpengine.f.rtpengine_offer the valid transport protocols are six: 
> RTP/AVP, RTP/SAVP, UDP/TLS/RTP/SAVP, RTP/AVPF, RTP/SAVPF, UDP/TLS/RTP/SAVPF - these serve as an alternative, more explicit way to select between the different RTP protocols and profiles supported by the RTP proxy. For example, giving the flag “RTP/SAVPF” has the same effect as giving the two flags “SRTP AVPF”.

But https://github.com/sipwise/rtpengine/blob/master/README.md mentions under `transport protocol` four protocols:
> Valid values are: RTP/AVP, RTP/AVPF, RTP/SAVP, RTP/SAVPF.

The values UDP/TLS/RTP/SAVPF and UDP/TLS/RTP/SAVP are in the source code and kamailio/rtpengine, but not in README.md

**Comment by dilyanpalauzov** [comment]
While at it, https://www.kamailio.org/docs/modules/devel/modules/rtpengine.html#rtpengine.f.rtpengine_offer has this option:

> force-answer - force “answer”, that is, only rewrite SDP when corresponding session already exists in the RTP proxy. By default is on when the session is to be completed. …      Check also the documentation of RTPEngine, these flags are documented there as well: https://github.com/sipwise/rtpengine.



Can you advice which command from the NG-protocols is switched, by using `force-answer` from Kamailio?

**Comment by rfuchs** [comment]
`force-answer` basically makes sure that when using `rtpengine_manage()` it does an `answer` command instead of an `offer`.

`rtpengine_manage()` is supposed to detect which one (offer or answer) is appropriate, but sometimes doesn't do what you want. With this flag you can make sure it does an `answer` and not an `offer`.

**Comment by dilyanpalauzov** [comment]
Thanks for your answer.  Can you elaborate why does `rtpengine_manage()` sometimes not recognize its role correctly?  If the Kamailio configurator has to tell `rtpengine_manage()` explicitly whether it is an answer or an offer, why not just using `rtpengine_answer()`/`rtpengine_offer()`?

**Comment by rfuchs** [comment]
Yeah you can do that. This is just another way to do the same thing.

**Comment by dilyanpalauzov** [comment]
Can you please document the cases, where `rtpengine_manage()` cannot determine whether it handles a request or an answer?

**Comment by rfuchs** [comment]
> Can you please document the cases, where `rtpengine_manage()` cannot determine whether it handles a request or an answer?

I can only speculate as we don't use `rtpengine_manage` ourselves, but probably one example would be when an answer is processed as part of a 200 OK message, when the corresponding offer (in the INVITE) wasn't handled via `rtpengine_manage` but rather via `rtpengine_offer`. In this case the internal flag `FL_SDP_BODY` would not be set and the module would think that this is an offer belonging to a 200 OK / ACK exchange.

**Comment by dilyanpalauzov** [comment]
https://www.kamailio.org/docs/modules/devel/modules/rtpengine.html#rtpengine.f.rtpengine_manage says:
> * If INVITE with SDP, when the tm module is loaded, mark transaction with internal flag FL_SDP_BODY to know that the 1xx and 2xx are for rtpengine_answer()
> * If reply with SDP to INVITE having code 1xx and 2xx, then do rtpengine_answer() if the request had SDP or tm is not loaded, otherwise do rtpengine_offer()

Should the second bullet be instead:

> If reply with SDP to INVITE having code 1xx and 2xx, then do rtpengine_answer() if flag FL_SDP_BODY is set or tm is not loaded, otherwise do rtpengine_offer()

The current wording indeed suggests that rtpengine_manage() does ANSWER, even if the offer was generated by rtpengine_offer() (so FL_SDP_BODY was not set).
